src/platform/portduino/PortduinoGlue.cpp

```diff
@@ -376,42 +360,65 @@ bool loadConfig(const char *configPath)
+                for (auto& loraModule : loraModules) {
+                    if (yamlConfig["Lora"]["Module"].as<std::string>("") == loraModule.strName) {
+                        settingsMap[loraModule.cfgName] = true;
+                        break;
+                    }
+                }
             }
+
             settingsMap[dio2_as_rf_switch] = yamlConfig["Lora"]["DIO2_AS_RF_SWITCH"].as<bool>(false);
             settingsMap[dio3_tcxo_voltage] = yamlConfig["Lora"]["DIO3_TCXO_VOLTAGE"].as<float>(0) * 1000;
             if (settingsMap[dio3_tcxo_voltage] == 0 && yamlConfig["Lora"]["DIO3_TCXO_VOLTAGE"].as<bool>(false)) {
                 settingsMap[dio3_tcxo_voltage] = 1800; // default millivolts for "true"
             }
-            settingsMap[cs] = yamlConfig["Lora"]["CS"].as<int>(RADIOLIB_NC);
-            settingsMap[irq] = yamlConfig["Lora"]["IRQ"].as<int>(RADIOLIB_NC);
-            settingsMap[busy] = yamlConfig["Lora"]["Busy"].as<int>(RADIOLIB_NC);
-            settingsMap[reset] = yamlConfig["Lora"]["Reset"].as<int>(RADIOLIB_NC);
-            settingsMap[txen] = yamlConfig["Lora"]["TXen"].as<int>(RADIOLIB_NC);
-            settingsMap[rxen] = yamlConfig["Lora"]["RXen"].as<int>(RADIOLIB_NC);
-            settingsMap[sx126x_ant_sw] = yamlConfig["Lora"]["SX126X_ANT_SW"].as<int>(RADIOLIB_NC);
-            settingsMap[gpiochip] = yamlConfig["Lora"]["gpiochip"].as<int>(0);
+
+            // backwards API compatibility and to globally set gpiochip once
+            int defaultGpioChip = settingsMap[default_gpiochip] = yamlConfig["Lora"]["gpiochip"].as<int>(0);
+
+            const struct { configNames pin;
+                           configNames gpiochip;
+                           configNames line;
+                           std::string strName; } pinMappings[] = {
+                { cs_pin, cs_gpiochip, cs_line, "CS" },
+                { irq_pin, irq_gpiochip, irq_line, "IRQ" },
+                { busy_pin, busy_gpiochip, busy_line, "Busy" },
+                { reset_pin, reset_gpiochip, reset_line, "Reset" },
+                { txen_pin, txen_gpiochip, txen_line, "TXen" },
+                { rxen_pin, rxen_gpiochip, rxen_line, "RXen" },
+                { sx126x_ant_sw_pin, sx126x_ant_sw_gpiochip, sx126x_ant_sw_line, "SX126X_ANT_SW" },
+            };
+            for (auto& pinMap : pinMappings) {
+                if (yamlConfig["Lora"][pinMap.strName].IsMap()
+                    && (yamlConfig["Lora"][pinMap.strName]["pin"]
+                      || yamlConfig["Lora"][pinMap.strName]["line"]
+                      || yamlConfig["Lora"][pinMap.strName]["gpiochip"])) {
+                    settingsMap[pinMap.pin] = yamlConfig["Lora"][pinMap.strName]["pin"].as<int>(RADIOLIB_NC);
+                    settingsMap[pinMap.line] = yamlConfig["Lora"][pinMap.strName]["line"].as<int>(settingsMap[pinMap.pin]);
+                    settingsMap[pinMap.gpiochip] = yamlConfig["Lora"][pinMap.strName]["gpiochip"].as<int>(defaultGpioChip);
```

Comment:
It would be great to include an example in the default config.yaml of this new approach.

Reply:
Added the example to the `bin/config-dist.yaml`